### DISCUSSION TITLE
V4 vs v3

### DISCUSSION THREAD

**Discussion by FRacine** [PSAppDeployToolkit/PSAppDeployToolkit#1132] [Q&A] [answered]
Hi,

I will not hide my concern for v4 of PSADT. I'm excited about v4 but worried.
Currently our toolkit scripts do a dot-source of "AppDeployToolkitMain.ps1" as well as our other function files. This has all worked very well for years. I'm probably not the only one.
1. We have never used “Deploy-Application.ps1”, is this a problem?
2. I understand that v4 is a module. I guess this means no longer dot-sourcing but installing a module at the beginning and removing it at the end. Am I correct? For the rest, will it work as before?
3. Why use frontline v3 or use V4? I'm guessing v3 won't be around forever.
4. What happen if v4 module is on the computer but you run an older package using a dot source v3?

Thanks,

**Comment by DanGough** [comment]
1. If you’ve never used the Deploy-Application.ps1, then no need to adopt it or the new version Invoke-AppDeployToolkit.ps1.

2. Instead of dot-sourcing, you just need to load in the module from a local folder via Import-Module. Alternatively you can install the module on the endpoints, then it will auto load whenever trying to use any of its commands. You will likely need to open a session also to activate logging and get access to various variables (https://psappdeploytoolkit.com/docs/adtsession-object).

3. The v3 template is there for those that want to be able to re-use their old Deployment-Application.ps1 scripts and use the old commands. The v4 template is for those that want to dive straight into using the new syntax and the new UI. I expect we’ll drop the v3 compatibility in future, perhaps for v5.

4. There should be no issues doing this.

**Reply by FRacine** [reply]
> 2. Instead of dot-sourcing, you just need to load in the module from a local folder via Import-Module. Alternatively you can install the module on the endpoints, then it will auto load whenever trying to use any of its commands. You will likely need to open a session also to activate logging and get access to various variables (https://psappdeploytoolkit.com/docs/adtsession-object).

So I will have to install the module, open the session, uninstall the module to do it clean?

> 3. The v3 template is there for those that want to be able to re-use their old Deployment-Application.ps1 scripts and use the old commands. The v4 template is for those that want to dive straight into using the new syntax and the new UI. I expect we’ll drop the v3 compatibility in future, perhaps for v5.
> 4. There should be no issues doing this.
Because you renamed the functions?

I believe somekind of documentations would be appropriate to clarify those steps. It is a major step.

**Reply by sintaxasn** [reply]
There's no need to install the module. Use Import-Module .\PSAppDeployToolkit.psd1 from within the module folder or simply Import-Module .\PSAppDeployToolkit from the parent folder.

As previously stated, we're heavily focused on documentation right now and we will be pushing updates over the course of the next few days.

**Reply by FRacine** [reply]
> There's no need to install the module. Use Import-Module .\PSAppDeployToolkit.psd1 from within the module folder or simply Import-Module .\PSAppDeployToolkit from the parent folder.
> 
> As previously stated, we're heavily focused on documentation right now and we will be pushing updates over the course of the next few days.

When you open a session which script is run to create the informations.?
But at the end we should uninstall the module to keep the computer clean?!

**Reply by mjr4077au** [reply]
From our Releases page, what you should be doing is downloading the Template_v3 file, dot-sourcing its `AppDeployToolkitMain.ps1` file, then testing your workflows with it.

You don't have that reinvent the wheel straight away, this compatibility layer exists to ease the transition for users like yourselves while you get used to the v4 experience, which in time you will see is significantly better.

The v3 compatibility layer won't be maintained by us forever, so don't just rely on it solely and call it a day.

**Reply by FRacine** [reply]
> When you open a session which script is run to create the informations.?
> But at the end we should uninstall the module to keep the computer clean?!

When you open a session which script is run to create the informations.?
But at the end we should uninstall the module to keep the computer clean?!

**Reply by mjr4077au** [reply, marked as answer]
![Screenshot_20241209-075850](https://github.com/user-attachments/assets/dd4ff649-b782-4d43-afc6-b5ac24451012)

Respectfully, please read the information that's been provided to you. Not once has it been said you need to install the module, and you've completely ignored everything I've said in my message to you. 

Read my message carefully, follow what it says, and you should be up and running with minimal fuss.

**Comment by DanGough** [comment]
For clarity, Import-Module does not install anything that needs to be cleaned up. Importing the module just loads it into memory, similar to dot sourcing the old toolkit.

I don’t understand your question about the session. But whichever script you write and run yourself should contain the session information.
